contracts/contracts/abstract/SelfVerificationRoot.sol

```diff
@@ -104,73 +130,102 @@ abstract contract SelfVerificationRoot is ISelfVerificationRoot {
      */
-    function _setScope(uint256 newScope) internal {
-        _scope = newScope;
+    function _setScope(uint256 _newScope) internal {
+        _scope = _newScope;
+        emit ScopeUpdated(_newScope);
     }
 
     /**
      * @notice Adds a new attestation ID to the allowed list
      * @dev Used to add support for additional attestation types
-     * @param attestationId The attestation ID to add
+     * @param _attestationId The attestation ID to add
      */
-    function _addAttestationId(uint256 attestationId) internal {
-        _attestationIds[attestationId] = true;
+    function _addAttestationId(uint256 _attestationId) internal {
+        _attestationIdToEnabled[_attestationId] = true;
+        emit AttestationIdAdded(_attestationId);
     }
 
     /**
      * @notice Removes an attestation ID from the allowed list
      * @dev Used to revoke support for specific attestation types
-     * @param attestationId The attestation ID to remove
+     * @param _attestationId The attestation ID to remove
      */
-    function _removeAttestationId(uint256 attestationId) internal {
-        _attestationIds[attestationId] = false;
+    function _removeAttestationId(uint256 _attestationId) internal {
+        _attestationIdToEnabled[_attestationId] = false;
+        emit AttestationIdRemoved(_attestationId);
     }
 
     /**
      * @notice Helper function to get an array of revealed data values from proof signals
      * @dev Returns an array of the three packed revealed data values
-     * @param pubSignals The proof's public signals
+     * @param _pubSignals The proof's public signals
      * @return revealedDataPacked Array of the three packed revealed data values
      */
     function getRevealedDataPacked(
-        uint256[21] memory pubSignals
+        uint256[21] calldata _pubSignals
     ) internal pure returns (uint256[3] memory revealedDataPacked) {
-        revealedDataPacked[0] = pubSignals[REVEALED_DATA_PACKED_INDEX];
-        revealedDataPacked[1] = pubSignals[REVEALED_DATA_PACKED_INDEX + 1];
-        revealedDataPacked[2] = pubSignals[REVEALED_DATA_PACKED_INDEX + 2];
+        revealedDataPacked[0] = _pubSignals[REVEALED_DATA_PACKED_INDEX];
+        revealedDataPacked[1] = _pubSignals[REVEALED_DATA_PACKED_INDEX + 1];
+        revealedDataPacked[2] = _pubSignals[REVEALED_DATA_PACKED_INDEX + 2];
         return revealedDataPacked;
     }
 
     /**
      * @notice Verifies a self-proof
      * @dev Validates scope and attestation ID before performing verification through the identity hub
-     * @param proof The proof data for verification and disclosure
+     * @param _proof The proof data for verification and disclosure
      */
-    function verifySelfProof(ISelfVerificationRoot.DiscloseCircuitProof memory proof) public virtual {
-        if (_scope != proof.pubSignals[CircuitConstants.VC_AND_DISCLOSE_SCOPE_INDEX]) {
+    function verifySelfProof(ISelfVerificationRoot.DiscloseCircuitProof calldata _proof) public {
```

Comment:
Maybe we should get bytes data for here to have forward compatibility in some id systems?